src/main/java/com/razorpay/PaymentClient.java

```diff
@@ -26,23 +29,19 @@ public Payment capture(String id, JSONObject request) throws RazorpayException {
     return post(String.format(Constants.PAYMENT_CAPTURE, id), request);
   }
 
-  public Refund refund(String id) throws RazorpayException {
```

Comment:
IMO, this should not be removed at all. Keeping id as a function parameter implicitly makes it a mandatory attribute. Also, since PaymentClient is meant to be a wrapper around the Payment resource the refund method exposed to the developer should resemble `/payment/:id/refund` and not `/refund`.